docs/pages/cairo/enum.md

```diff
@@ -2,53 +2,60 @@
 
 ## What is Enum
 Enums, short for "enumerations," are a way to define a custom data type that consists of a fixed set of named values, called variants. Enums are useful for representing a collection of related values where each value is distinct and has a specific meaning. Enums are particularly useful in game development for representing game states, player actions, or any other set of related constants that a game might need to track.
+
+In this example, we've defined an enum called PlayerCharacter with four variants: `Godzilla`, `Dragon`, `Fox`, and `Rhyno`. The naming convention is to use `PascalCase` for enum variants. Each variant represents a distinct value of the `PlayerCharacter` type. In this particular example, variants don't have any associated value. 
+Now let's imagine that our variants have associated values, We can define a new PlayerCharacter enum:
+
 ```rust
 # [derive(Serde, Copy, Drop, Introspect, PartialEq, Print)]
 enum PlayerCharacter {
-    Godzilla,
-    Dragon,
-    Fox,
-    Rhyno
+    Godzilla: u128,
+    Dragon:   u128,
+    Fox:      u128,
+    Rhyno:    u128
 }
 
 ```
-In this example, we've defined an enum called PlayerCharacter with four variants: `Godzilla`, `Dragon`, `Fox`, and `Rhyno`. The naming convention is to use PascalCase for enum variants. Each variant represents a distinct value of the PlayerCharacter type. In this particular example, variants don't have any associated value. 
-Now let's imagine that our variants have associated values, We can define a new PlayerCharacter enum:
+However, in Dojo, all enum variants must share the same type. This limitation poses a challenge when we want to associate different data types with each variant, as is often the case in game development where characters, items, or enemies may have unique attributes.
+To work around this limitation, we can use a struct to encapsulate the attributes of each character type. This allows us to maintain the unique characteristics of each character while adhering to the constraints.
 
 ```rust
 # [derive(Serde, Copy, Drop, Introspect, PartialEq, Print)]
 enum PlayerCharacter {
-    Godzilla :u128,
-    Dragon : u128,
-    Fox : u128,
-    Rhyno : u128
+    Godzilla(CharacterAttributes),
+    Fox(CharacterAttributes),
+    Rhyno(CharacterAttributes),
```

Comment:
Super nice, elegant solution to showcase how this can be done. 👍

One the dojo storage layout will be re-work, we should be able to support natively enums with different types.